packages/vite/src/node/plugins/index.ts

```diff
@@ -60,13 +60,14 @@ export async function resolvePlugins(
       isBuild
     ),
     wasmPlugin(config),
-    webWorkerPlugin(config),
-    workerImportMetaUrlPlugin(config),
     assetPlugin(config),
     ...normalPlugins,
     definePlugin(config),
     cssPostPlugin(config),
     config.build.ssr ? ssrRequireHookPlugin(config) : null,
+    isBuild && buildHtmlPlugin(config),
+    webWorkerPlugin(config),
```

Comment:
I don't think we should move this plugin at least, normalPlugins are going to end up before it instead of after it and this could change user expectations. Assets plugins are between pre and normal plugins right now.

Reply:
I also think of a situation that is a normal html script tag, but this tag does not actually execute these two plugins.